From: Borislav Petkov <bp@suse.de>
Date: Thu, 15 Sep 2016 02:35:29 +0100
Subject: Revert fix minor infoleak in get_user_ex()
Patch-mainline: no, this one should be replaced with the -stable 4.1 revert
References: bsc#1008650 CVE-2016-9178

Revert upstream commit 1c109fabbd51 ("fix minor infoleak in get_user_ex()")

See 1008650 - CVE-2016-9178 for more info.

Signed-off-by: Borislav Petkov <bp@suse.de>

diff --git a/arch/x86/include/asm/uaccess.h b/arch/x86/include/asm/uaccess.h
index d081e7e42fb3..ace9dec050b1 100644
--- a/arch/x86/include/asm/uaccess.h
+++ b/arch/x86/include/asm/uaccess.h
@@ -391,11 +391,7 @@ do {									\
 #define __get_user_asm_ex(x, addr, itype, rtype, ltype)			\
 	asm volatile("1:	mov"itype" %1,%"rtype"0\n"		\
 		     "2:\n"						\
-		     ".section .fixup,\"ax\"\n"				\
-                     "3:xor"itype" %"rtype"0,%"rtype"0\n"		\
-		     "  jmp 2b\n"					\
-		     ".previous\n"					\
-		     _ASM_EXTABLE_EX(1b, 3b)				\
+		     _ASM_EXTABLE_EX(1b, 2b)				\
 		     : ltype(x) : "m" (__m(addr)))
 
 #define __put_user_nocheck(x, ptr, size)			\
